.github/workflows/lint-test.yaml

```diff
@@ -0,0 +1,30 @@
+name: Lint
```

Comment:
I would move this GitHub action under `test-python.yaml`,  with the similar name as here: https://github.com/kubeflow/trainer/blob/master/.github/workflows/test-python.yaml#L1

In that test, we can just run `make verify` which verifies Python code, and run `make test-unit` which runs unit tests.

Removing number of parallel GitHub actions will help our CI to execute faster, given that we have limitations for concurrent runs.